include/cone_detector/cone_detector.h

```diff
@@ -134,11 +149,16 @@ class ConeDetector {
 	bool hullIsValid(std::vector<cv::Point>& hull);
 
 	// Process one image.
+	cv::Mat last_image_;
+	long last_image_count_;
 	void imageCb(cv::Mat& image);
 
 	// Process one image topic message.
 	void imageTopicCb(const sensor_msgs::ImageConstPtr& msg);
 
+	// Compute KMEANS on image.
+	void kmeansImage(cv::Mat image);
```

Comment:
There was a direct modification of the image in the code. I made a local copy now and made the param const and by reference. Eventually, this block of code will do more than write interesting debugger output.